omnigibson/objects/object_base.py

```diff
@@ -365,29 +370,30 @@ def get_base_aligned_bbox(self, link_name=None, visual=False, xy_aligned=False):
             # To find the rotation that this transform does around the Z axis, we rotate the [1, 0, 0] vector by it
             # and then take the arctangent of its projection onto the XY plane.
             rotated_X_axis = base_frame_to_world[:3, 0]
-            rotation_around_Z_axis = np.arctan2(rotated_X_axis[1], rotated_X_axis[0])
-            xy_aligned_base_com_to_world = trimesh.transformations.compose_matrix(
-                translate=translate, angles=[0, 0, rotation_around_Z_axis]
+            rotation_around_Z_axis = th.arctan2(rotated_X_axis[1], rotated_X_axis[0])
+            xy_aligned_base_com_to_world = th.tensor(
+                trimesh.transformations.compose_matrix(translate=translate, angles=[0, 0, rotation_around_Z_axis]),
+                dtype=th.float32,
             )
```

Comment:
maybe we should create a torch version of this in `transform_utils`. let's try to get rid of all `trimesh.transforms` calls?